### PR TITLE
Fix recheck and logDownloads setters Add internet recheck option to GUI

### PR DESCRIPTION
**Pull request by selgamal**
## Description
1. Fix WebCache properties recheck and logDownloads setters.
2. Add option to control internet recheck intervals in GUI.

### 1. WebCache recheck and logDownloads properties setters: 
`WebCache.recheck` and `WebCache.logDownLoads` not returning the correct value. The issue can be demonstrated and tested as follows:
```python
from arelle import Cntlr, FileSource

cntlr = Cntlr.Cntlr(logFileName="logToPring")

## before commit 
### set internet recheck and logDownload
cntlr.webCache.recheck = 'monthly'
cntlr.webCache.logDownloads = True
print(cntlr.webCache.recheck, cntlr.webCache.maxAgeSeconds) 
# >>> None 2592000.0 -> expected: monthly 2592000.0
print(cntlr.webCache.logDownloads)
# >>> None  -> expected True

# now set timeout
cntlr.webCache.timeout = 10
print(cntlr.webCache.recheck, cntlr.webCache.logDownloads, cntlr.webCache.timeout)
# >>> 10 10 10 -> expected: monthly True 10
# max age seconds still correct
print(cntlr.webCache.maxAgeSeconds)
# >>> 2592000.0

## after commit 
### set internet recheck and logDownload
cntlr.webCache.recheck = 'monthly'
cntlr.webCache.logDownloads = True
print(cntlr.webCache.recheck, cntlr.webCache.maxAgeSeconds) 
# >>> monthly 2592000.0 -> as expected
print(cntlr.webCache.logDownloads)
# >>> True  -> as expected

# set timeout
cntlr.webCache.timeout = 10
print(cntlr.webCache.recheck, cntlr.webCache.logDownloads, cntlr.webCache.timeout)
# >>> monthly True 10 -> as expected
# max age seconds still correct
print(cntlr.webCache.maxAgeSeconds)
# >>> 2592000.0
```
### 2. Add internet recheck interval to GUI:
Nice to have feature to be able to control recheck intervals from GUI since it is already available in command line. 
Added menu under `Tools>Internet>Internet recheck interval`:
recheck_opts
When working offline, selection is disabled:
recheck_opts2

**Added configuration key `internetRecheck` to save and restore selection.**

**Testing**: Set Show debug messages from `Tools>Messages log>Show debug messages` and change selections, the new values for `WebCache.recheck` and `WebCache.maxAgeSeconds` will be logged in the messages panel.
Also can redo same test from [here](https://github.com/Arelle/Arelle/issues/139#issuecomment-1047420213) but make sure to restart arelle after changing dates in `cachedUrlCheckTimes.json` so it will take effect.

### CODE REVIEW

File: arelle/CntlrWinMain.py [hunk 1]
```diff
@@ -201,12 +201,30 @@ def __init__(self, parent):
...
+        _recheck_initial = 'disable' if self.webCache.workOffline else 'normal'
+        self.internetRecheckVar.trace("w", self.setInternetRecheck)
+
+        _internetRecheckEntries = ((_("daily"), "daily"), (_("weekly"), "weekly"
+                                ), (_("monthly"), "monthly"), (_("never"), "never"))
+
+        for (_opt_label, _opt_val) in _internetRecheckEntries:
+            internetCacheRecheckMenu.add_checkbutton(label=_opt_label,
+                    variable=self.internetRecheckVar, underline=0,
+                    onvalue=_opt_val)
```
Comment (derekgengenbacher-wf) [inline arelle/CntlrWinMain.py:218]:
```suggestion
            internetCacheRecheckMenu.add_checkbutton(
                    label=_opt_label,
                    variable=self.internetRecheckVar, 
                    underline=0,
                    onvalue=_opt_val
             )
```

File: arelle/CntlrWinMain.py [hunk 2]
```diff
@@ -1088,7 +1106,21 @@ def setWorkOffline(self, *args):
...
+        else:
+            self._cacheMenu.entryconfig(self._internetRecheckLabel, state='normal')
+
+    def setInternetRecheck(self, *args):
+        self.webCache.recheck = self.internetRecheckVar.get()
+        self.config["internetRecheck"] = self.webCache.recheck
+        self.addToLog('WebCache.recheck = {}'.format(self.webCache.recheck),
+                    messageCode='debug', level=logging.DEBUG)
+        self.addToLog('WebCache.maxAgeSeconds = {}'.format(self.webCache.maxAgeSeconds),
+                    messageCode='debug', level=logging.DEBUG)
```
Comment (derekgengenbacher-wf) [inline arelle/CntlrWinMain.py:1121]: This addToLog and the one line above should fit onto one line.

**Comment by derekgengenbacher-wf** [comment]
@selgamal Thanks for the PR. Functionality wise it looks good. Just some python formatting stuff to fix.

**Comment by selgamal** [comment]
@derekgengenbacher-wf  updated formatting

**Comment by austinmatherne-wk** [comment]
Just a heads up, this PR isn't opened against master

**Comment by selgamal** [comment]
> Just a heads up, this PR isn't opened against master



> Just a heads up, this PR isn't opened against master

Should I reopen this request against master? I thought it will be better to keep all changes for this issue in same branch

**Comment by sagesmith-wf** [comment]
@selgamal yes, suggest closing this and opening against master

**Comment by selgamal** [comment]
@sagesmith-wf changed base branch to master

**Review by austinmatherne-wk — APPROVED** [review]

**Review by sagesmith-wf — APPROVED** [review]

**Review by derekgengenbacher-wf — APPROVED** [review]

**Review by hermfischer-wf — APPROVED** [review]

**Pull request merged**